Adding s3 bucket sink

matzew opened the pull request:
# Changes

<!-- 
Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! 

- :gift: Add new feature
- :bug: Fix bug
- :broom: Update or clean up current behavior
- :wastebasket: Remove feature or internal logic
-->

- as per title

<!--
In addition, categorize the changes you're making using the "/kind" Prow command, example:

/kind <kind>

Supported kinds are: api-change, bug, cleanup, deprecation, removal, documentation, enhancement, performance

-->
/kind <kind>

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

<!-- Please include the 'why' behind your changes if no issue exists -->

**Release Note**

<!--
:page_facing_up: If this change has user-visible impact, write a release note in the block
below. Include the string "action required" if additional action is required of
users switching to the new release, for example in case of a breaking change.

Write as if you are speaking to users, not other Knative contributors. If this
change has no user-visible impact, no release-note is needed.
-->
```release-note

```

**Docs**

<!--
:book: If this change has user-visible impact, link to an issue or PR in
https://github.com/knative/docs.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [knative/docs]: <issue or pr link>
- [Feature Track]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

christophd commented on aws-s3-sink/pom.xml:
```diff
@@ -0,0 +1,167 @@
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.eclipse.jetty.ee10</groupId>
```
The dependency is not required anymore as we have pinned the version in dependencyManagement in parent POM

christophd commented on aws-s3-sink/src/main/resources/application.properties:
```diff
@@ -0,0 +1,40 @@
+# Knative Camel component
+camel.component.knative.environmentPath=classpath:knative.json
+
+camel.component.knative.ceOverride[ce-type]={{kn.connector.ce.override.type:dev.knative.connector.event.aws-s3}}
```
these ceOverrides are not required for the sink

christophd commented on aws-s3-sink/src/main/resources/camel/kn-connector-source.yaml:
```diff
@@ -0,0 +1,35 @@
+metadata:
+  name: kn-connector-aws-s3-sink
+spec:
+  source:
```
source should be the Knative Broker

christophd commented on aws-s3-sink/src/main/resources/camel/kn-connector-source.yaml:
```diff
@@ -0,0 +1,35 @@
+      apiVersion: camel.apache.org/v1
+      kind: Kamelet
+      name: aws-s3-sink
+  sink:
```
sink should be the aws-s3-sink Kamelet

christophd reviewed (commented):
The sink-connector has mixed a few things coming from a source-connector (e.g. in kn-connector-source.yaml which should rather follow the concepts of kn-connector-sink.yaml). 

Please have a look into log-sink connector as an example or use the Maven archetype for sink connectors to generate the project

```
mvn archetype:generate \
 -DarchetypeGroupId=dev.knative.eventing \
 -DarchetypeArtifactId=kn-connector-archetype-sink \
 -DarchetypeVersion=1.0-SNAPSHOT
```

matzew commented:
I used the archetype now as:

```
mvn archetype:generate \
 -DarchetypeGroupId=dev.knative.eventing \
 -DarchetypeArtifactId=kn-connector-archetype-sink \
 -DarchetypeVersion=1.0-SNAPSHOT \
 -DartifactId=aws-s3-sink \
 -Dkamelet-name=aws-s3 \
 -Dcomponent-name=aws2-s3
```

the `pom.xml` diff is manually - I just added the sink module.

I reverted the change from the archetype since it was adding new lines after each existing line. Was very verbose

christophd commented:
> I reverted the change from the archetype since it was adding new lines after each existing line. Was very verbose
yes, Maven archetype process goes wild on formatting the existing POM. Revert and adding the module manually is the best option for now. thanks

christophd commented on aws-s3-sink/properties.adoc:
```diff
@@ -0,0 +1,4 @@
+|===
+|Property |Required |EnvVar |Description
```
consider running `mvn package` once before the PR, so the docs get generated

christophd reviewed (approved):
LGTM now
